strings/base_delegate.h

```diff
@@ -6,39 +6,63 @@ namespace winrt::impl
 #pragma warning(disable:4458) // declaration hides class member (okay because we do not use named members of base class)
 #endif
 
-    template <typename T, typename H>
-    struct implements_delegate : abi_t<T>, H, update_module_lock
+    template<bool try_marshaling> struct implements_delegate_base
```

Comment:
The `variadic_delegate` doesn't need to implement `IMarshal` but if it makes things simpler (by de-templatizing this) we could just do so conistently and avoid the split.